[TextFields] Support clear button tint color

romoore opened the pull request:
Adding support to change the clearButton's tintColor from the TextInputController. This allows more consistent theming across controllers.

Closes #4447 

## Filled

**Before**
![screen shot 2018-06-29 at 12 23 12 pm](https://user-images.githubusercontent.com/1753199/42103490-427a1e96-7b97-11e8-9410-9d20f5ff6d32.png)


**After**
![screen shot 2018-06-29 at 12 21 51 pm](https://user-images.githubusercontent.com/1753199/42103502-4b1d8c22-7b97-11e8-85be-3997d0e3dee4.png)


## Outlined

**Before**
![screen shot 2018-06-29 at 12 23 01 pm](https://user-images.githubusercontent.com/1753199/42103497-45b2fca4-7b97-11e8-82df-3a01bc3e8240.png)

**After**
![screen shot 2018-06-29 at 12 21 36 pm](https://user-images.githubusercontent.com/1753199/42103498-48b22010-7b97-11e8-8c8e-004cb35d3b6a.png)

Commit:
Initial work

Commit:
[TextFields] Customize clear button tint color

TextFields have a clear button with a default gray color. To support
broader customization, TextInputControllers should support changing this
color.

Commit:
Finishing tests, trying to fix backgroundColor behavior.

* Placeholder label font in Filled is still changes sizes when text is
  input.
* Need to review how Outlined works

Commit:
Reverting changes around `backgroundColor`

Commit:
Improving clear button contrast in themers.

Commit:
Fixing white space

material-automation commented:
## API diff detected the following changes

### TextFields

#### MDCTextInputControllerBase

*new* property: `textInputClearButtonTintColorDefault` in `MDCTextInputControllerBase`

*new* property: `textInputClearButtonTintColor` in `MDCTextInputControllerBase`

<!-- identifier: apidiff -->

willlarche commented on components/TextFields/src/MDCTextInputControllerBase.h:
```diff
@@ -104,4 +104,15 @@ extern const CGFloat MDCTextInputControllerBaseDefaultBorderRadius;
  */
 @property(nonatomic, assign) NSUInteger minimumLines;
 
+/**
```
I think this should be universal. Put it in `MDCTextInputController`.

romoore replied:
We spoke about this offline before I pushed up the PR. You agreed that we should only make the change for Outlined and Filled styles at this point.  The reasoning, as I understood it, is that [Material Guidelines only recommend the Outlined and Filled styles](https://material.io/design/components/text-fields.html#usage) based on internal usability testing.

Has something changed so that we should be actively supporting the other styles with new feature development?

romoore replied:
After discussing, I've filed #4453 as a follow-up to migrate this property up to the root protocol. Due to the amount of testing that should be done across implementing classes, we decided not to add it to this PR.

Commit:
Restore 54% opacity

Commit:
Fixing icon drawing code.

Commit:
Reverting clearButton drawing code.

Commit:
Fix formatting

Commit:
Removing backgroundColor assignment.

`backgroundColor` does nothing except in FullWidth controllers. Removing the
assignment in the themers to avoid confusion.

Commit:
Merge branch 'develop' into issue-4447

willlarche commented on components/TextFields/examples/TextFieldOutlinedExample.m:
```diff
@@ -38,9 +43,26 @@ - (void)dealloc {
   [super viewDidLoad];
-  self.view.backgroundColor = [UIColor whiteColor];
+
+  if (!self.colorScheme) {
```
`viewDidLoad` is guaranteed to be called once. I think you can drop these null checks.

romoore replied:
It's in case the schemes are injected by the Catalog, we don't want to overwrite those with the defaults.

willlarche reviewed (approved):
With comment.

Commit:
Merge branch 'develop' into issue-4447
